File: dotnet/Dilithia.Sdk/Models/CredentialSchema.cs
namespace Dilithia.Sdk.Models;

/// <summary>
/// A credential schema definition registered on-chain.
/// </summary>
/// <param name="Id">The unique schema identifier.</param>
/// <param name="Name">The human-readable schema name.</param>
/// <param name="Issuer">The address of the schema registrant.</param>
/// <param name="Fields">The list of field names defined in the schema.</param>
public record CredentialSchema(string Id, string Name, Address Issuer, List<string> Fields);

File: java/src/main/java/org/dilithia/sdk/model/CredentialSchema.java
package org.dilithia.sdk.model;

import java.util.List;
import java.util.Map;

/**
 * Represents a credential schema registered on-chain.
 *
 * @param name       the schema name
 * @param version    the schema version
 * @param attributes the list of attribute definitions (each a map with "name" and "type" keys)
 */
public record CredentialSchema(String name, String version, List<Map<String, String>> attributes) {}
